tests/codegen/string_equal_comparison.cpp

```diff
@@ -36,95 +30,51 @@ define i64 @"kretprobe:vfs_read"(i8* nocapture readnone) local_unnamed_addr sect
-  call void @llvm.memcpy.p0i8.p0i8.i64(i8* nonnull align 1 %3, i8* nonnull align 1 %4, i64 16, i1 false)
-  %pseudo = call i64 @llvm.bpf.pseudo(i64 1, i64 1)
-  %lookup_elem = call i8* inttoptr (i64 1 to i8* (i8*, i8*)*)(i64 %pseudo, [16 x i8]* nonnull %"@_key")
-  %map_lookup_cond = icmp eq i8* %lookup_elem, null
-  br i1 %map_lookup_cond, label %lookup_merge, label %lookup_success
-
 strcmp.loop:                                      ; preds = %entry
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %strcmp.char2)
-  %5 = add [16 x i8]* %comm, i64 1
-  %probe_read3 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char2, i64 1, [16 x i8]* %5)
-  %6 = load i8, i8* %strcmp.char2, align 1
-  %strcmp.cmp4 = icmp eq i8 %6, 115
+  %3 = add [16 x i8]* %comm, i64 1
+  %probe_read3 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char2, i64 1, [16 x i8]* %3)
+  %4 = load i8, i8* %strcmp.char2, align 1
+  %strcmp.cmp4 = icmp eq i8 %4, 115
   br i1 %strcmp.cmp4, label %strcmp.loop1, label %pred_false
 
 strcmp.loop1:                                     ; preds = %strcmp.loop
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %strcmp.char6)
-  %7 = add [16 x i8]* %comm, i64 2
-  %probe_read7 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char6, i64 1, [16 x i8]* %7)
-  %8 = load i8, i8* %strcmp.char6, align 1
-  %strcmp.cmp8 = icmp eq i8 %8, 104
+  %5 = add [16 x i8]* %comm, i64 2
+  %probe_read7 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char6, i64 1, [16 x i8]* %5)
+  %6 = load i8, i8* %strcmp.char6, align 1
+  %strcmp.cmp8 = icmp eq i8 %6, 104
   br i1 %strcmp.cmp8, label %strcmp.loop5, label %pred_false
 
 strcmp.loop5:                                     ; preds = %strcmp.loop1
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %strcmp.char10)
-  %9 = add [16 x i8]* %comm, i64 3
-  %probe_read11 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char10, i64 1, [16 x i8]* %9)
-  %10 = load i8, i8* %strcmp.char10, align 1
-  %strcmp.cmp12 = icmp eq i8 %10, 100
+  %7 = add [16 x i8]* %comm, i64 3
+  %probe_read11 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char10, i64 1, [16 x i8]* %7)
+  %8 = load i8, i8* %strcmp.char10, align 1
+  %strcmp.cmp12 = icmp eq i8 %8, 100
   br i1 %strcmp.cmp12, label %strcmp.loop9, label %pred_false
 
 strcmp.loop9:                                     ; preds = %strcmp.loop5
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %strcmp.char14)
-  %11 = add [16 x i8]* %comm, i64 4
-  %probe_read15 = call i64 inttoptr (i64 4 to i64 (i8*, i64, i8*)*)(i8* nonnull %strcmp.char14, i64 1, [16 x i8]* %11)
-  %12 = load i8, i8* %strcmp.char14, align 1
-  %strcmp.cmp16 = icmp eq i8 %12, 0
-  br i1 %strcmp.cmp16, label %pred_true, label %pred_false
-
-lookup_success:                                   ; preds = %pred_true
```

Comment:
Why is this whole section dissapearing? That can't be ok

Reply:
```
vagrant@ubuntu-cosmic:~/bpftrace$ sudo bpftrace_master -e 't:syscalls:sys_enter_openat /comm == "cat"/ { @[comm]++ }' -c "cat xxx"
Attaching 1 probe...
/bin/cat: xxx: No such file or directory


@[cat]: 25

vagrant@ubuntu-cosmic:~/bpftrace$ sudo bpftrace -e 't:syscalls:sys_enter_openat /comm == "cat"/ { @[comm]++ }' -c "cat xxx"
Attaching 1 probe...
/bin/cat: xxx: No such file or directory
```

WIth your patch the whole function body seems to be optimized away. The optimized IR doesn't contain the map lookup and store code